zebra-state/src/config.rs

```diff
@@ -317,6 +317,15 @@ fn check_and_delete_database(
         return None;
     }
 
+    // Don't delete databases that can be reused.
```

Comment:
There are some extra checks in `try_reusing_previous_db_after_major_upgrade()` that would be nice to have here as well.

Reply:
Do you have any specific ones in mind? If yes, let's add them to this PR, otherwise let's do it in a future one?